nengo/conftest.py

```diff
@@ -391,24 +392,34 @@ def pytest_runtest_setup(item):  # noqa: C901
     rc.set('decoder_cache', 'enabled', 'False')
     rc.set('exceptions', 'simplified', 'False')
 
-    if not hasattr(item, 'obj'):
-        return  # Occurs for doctests, possibly other weird tests
-
-    test_uses_sim = 'Simulator' in item.fixturenames
-    test_uses_refsim = 'RefSimulator' in item.fixturenames
-    tests_frontend = not (test_uses_sim or test_uses_refsim)
-
-    if not tests_frontend:
-        item_name = get_item_name(item)
-
-        for test, reason in TestConfig.Simulator.unsupported:
-            # We add a '*' before test to eliminate the surprise of needing
-            # a '*' before the name of a test function.
-            if fnmatch(item_name, '*' + test):
-                if TestConfig.run_unsupported:
-                    item.add_marker(pytest.mark.xfail)
-                else:
-                    pytest.skip(reason)
+    item_name = get_item_name(item)
+
+    # join all the lines and then split (preserving quoted strings)
+    unsupported = shlex.split(
+        " ".join(item.config.getini("nengo_test_unsupported")))
+    # group pairs (representing testname + reason)
+    unsupported = [
+        unsupported[i:i + 2] for i in range(0, len(unsupported), 2)]
+
+    for test, reason in unsupported:
+        # escape brackets
+        escaped = ""
+        for c in list(test):
+            if c in ("[", "]"):
+                escaped += "[" + c + "]"
+            else:
+                escaped += c
+        test = escaped
```

Comment:
I was thinking we could use `.replace` for this, then I realized since you're replacing square brackets with groups of square brackets its hard not to have the second replace modify the first replace. Anyway, made a more concise expression for this loop, and added more comment, which I think is clearer. Was there any reason for using `list(test)` here instead of just looping over `test`?

Reply:
I just copied this from some older code I had, I don't actually remember why I did it that way.  Maybe for 2.7 compatibility (which wouldn't be an issue anymore)?  Or maybe it's just old code that wasn't the best haha.